File: WeatherForecast/WeatherForecast/Models/City.m
//
//  City.m
//  WeatherForecast
//
//  Created by Hakkı Yiğit Yener on 23.05.2020.
//  Copyright © 2020 N11.com. All rights reserved.
//

#import "City.h"

#define kNameKey        @"Name"
#define kDayCountKey    @"DayCount"
@implementation City

- (void)encodeWithCoder:(nonnull NSCoder *)coder {
    [coder encodeObject:self.name    forKey:kNameKey];
    [coder encodeInt:self.reportDayCount forKey:kDayCountKey];
}

- (nullable instancetype)initWithCoder:(nonnull NSCoder *)coder {
    self.name = [coder decodeObjectForKey:kNameKey];
    self.reportDayCount = [coder decodeIntForKey:kDayCountKey];
    return self;
}
@end

File: WeatherForecast/WeatherForecast/Models/City.swift
//
//  CitySwift.swift
//  WeatherForecast
//
//  Created by Hakkı Yiğit Yener on 27.05.2020.
//  Copyright © 2020 N11.com. All rights reserved.
//

import UIKit

public class City: NSObject,NSCoding {
    @objc public var name: String? = nil
    @objc public var reportDayCount: Int32 = 0
    
    @objc public override init() {
        
    }
    required public init?(coder: NSCoder) {
        name = coder.decodeObject(forKey: "name") as? String
        reportDayCount = coder.decodeInt32(forKey: "reportDayCount")
    }
    
    public func encode(with coder: NSCoder) {
        coder.encode(name, forKey: "name")
        coder.encode(reportDayCount, forKey: "reportDayCount")
    }
    
    


}
